### PR TITLE
Disable `opcache.preload` if it's enabled

### PR DESCRIPTION
**Pull request by weirdan**
Fixes vimeo/psalm#9382

Also reorganized the opcache settings check a bit and enforced
`opcache.jit_buffer_size`

### CODE REVIEW

**Comment by weirdan** [comment]
I've created 5.7.x branch so that we can merge this PR without necessarily releasing 5.8

File: src/Psalm/Internal/Fork/PsalmRestarter.php [hunk 1]
```diff
@@ -53,28 +63,54 @@ protected function requiresRestart($default): bool
...
+                'enable_cli' => in_array(ini_get('opcache.enable_cli'), ['1', 'true', true, 1]),
+                'jit' => (int) ini_get('opcache.jit'),
+                'optimization_level' => (string) ini_get('opcache.optimization_level'),
+                'preload' => (string) ini_get('opcache.preload'),
+                'jit_buffer_size' => self::toBytes(ini_get('opcache.jit_buffer_size')),
+            ];
 
-            if (((int) ini_get('opcache.jit')) !== 1205) {
-                return true;
+            foreach (self::REQUIRED_OPCACHE_SETTINGS as $ini_name => $required_value) {
```
Comment (danog) [inline src/Psalm/Internal/Fork/PsalmRestarter.php:78]: Couldn't this be just an `if (self::REQUIRED_OPCACHE_SETTINGS !== $opcache_settings) { return true; }`?

**Review by danog — CHANGES_REQUESTED** [review]
LGTM!

File: src/Psalm/Internal/Fork/PsalmRestarter.php (hunk 1 above)
Comment (weirdan) [reply src/Psalm/Internal/Fork/PsalmRestarter.php:78]: Perhaps, but that would require the keys in both arrays to be in the same order, which could easily be overlooked when adding/removing items.

Comment (danog) [reply src/Psalm/Internal/Fork/PsalmRestarter.php:78]: Makes sense!

**Pull request merged**